public/app/features/scenes/dashboard/ShareQueryDataProvider.ts

```diff
@@ -0,0 +1,100 @@
+import { Unsubscribable } from 'rxjs';
+
+import {
+  SceneDataProvider,
+  SceneDataState,
+  SceneDataTransformer,
+  SceneObject,
+  SceneObjectBase,
+  SceneQueryRunner,
+} from '@grafana/scenes';
+import { DashboardQuery } from 'app/plugins/datasource/dashboard/types';
+
+export interface ShareQueryDataProviderState extends SceneDataState {
+  query: DashboardQuery;
+}
+
+export class ShareQueryDataProvider extends SceneObjectBase<ShareQueryDataProviderState> implements SceneDataProvider {
+  private _querySub: Unsubscribable | undefined;
+
+  public constructor(state: ShareQueryDataProviderState) {
+    super(state);
+
+    this.addActivationHandler(() => {
+      // TODO handle changes to query model (changed panelId / withTransforms)
+      //this.subscribeToState(this._onStateChanged);
+      this.subscribeToSource();
+    });
+  }
+
+  public subscribeToSource() {
+    const { query } = this.state;
+
+    if (this._querySub) {
+      this._querySub.unsubscribe();
+    }
+
+    if (!query.panelId) {
+      return;
+    }
+
+    const keyToFind = `panel-${query.panelId}`;
+    const source = findObjectInScene(this.getRoot(), (scene: SceneObject) => scene.state.key === keyToFind);
+
+    if (!source) {
+      console.log('Shared dashboard query refers to a panel that does not exist in the scene');
+      return;
+    }
+
+    const sourceData = source.state.$data;
+    if (!sourceData) {
+      console.log('No source data found for shared dashboard query');
+      return;
+    }
+
+    if (!sourceData?.isActive) {
+      // TODO, not sure the best way to handle this, how to handle deactivation of the source when we activated it?
+      sourceData?.activate();
```

Comment:
@torkelo I think i have faced the same issue when tried the `inputParams` approach. What I did there I had a references map in the SceneObjectBase, and only actually de-activated the object if the last reference was de-activating it.

In this case the source data is pretty much an implicit input param. Maybe we could try approaching this in a similar way too. Guess `activate` would require type change to accept a consumers key. And given that we have activation handlers now - and were thinking about making `activate` private - maybe we can change the usage of the activeate to actually support the cases like in this PR? Then we would have:
- `private activate` - for self-activation handling
- `addActivationHandler` - for objects interactions handling
- `public activateFrom(source)` - for external activation handling, subject of reference counting

Reply:
I wonder if TS allows two methods (public/ and private) with the same name but different signatture...